recipes/dpp/all/conanfile.py

```diff
@@ -0,0 +1,95 @@
+import os
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.build import check_min_cppstd
+from conan.tools.cmake import CMakeToolchain, CMake, cmake_layout, CMakeDeps
+from conan.tools.files import get, copy, rmdir
+from conan.tools.scm import Version
+
+required_conan_version = ">=1.54"
+
+class DPPConan(ConanFile):
+    name = "dpp"
+    license = "Apache-2.0"
+    package_type = "shared-library"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/brainboxdotcc/DPP"
+    description = "D++ is a lightweight and efficient library for Discord"
+    topics = ("discord")
+    settings = "os", "compiler", "build_type", "arch"
+
+    @property
+    def _min_cppstd(self):
+        return 17
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "apple-clang": "14",
+            "clang": "10",
+            "gcc": "8",
+            "msvc": "191",
+            "Visual Studio": "16",
+        }
+
+    def validate(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            check_min_cppstd(self, self._min_cppstd)
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(
+                f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+            )
+
+    def requirements(self):
+        self.requires("nlohmann_json/3.11.2")
```

Comment:
```suggestion
        self.requires("nlohmann_json/3.11.2", transitive_libs=True, transitive_headers=True)
```

nlohmann_json is exposed in public headers. Conan 2.x makes all dependencies as private by default. Need to tell Conan to expose nlohmann_json headers and make any definition as public to the downstream